tools/indexer/main.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type packageSet map[string]bool
+
+// adds any missing SDK packages to godoc.org
+func main() {
+	dir := ""
+	if len(os.Args) > 1 {
+		// assume second arg is source dir
+		dir = os.Args[1]
+	} else {
+		// if no args specified assume we're running from the source dir
+		// and calculate the relative path to the services directory.
+		var err error
+		dir, err = filepath.Abs("../../services")
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	pkgs, err := getPackagesForIndexing(dir)
+	if err != nil {
+		panic(err)
+	}
+
+	indexedPkgs, err := getIndexedPackages()
+	if err != nil {
+		panic(err)
+	}
+
+	// for each package in pkgs, check if it's already been
+	// indexed.  if it hasn't been indexed then do so
+
```

Comment:
maybe add a comment that indexing will happen if there is a get on the package or/and why is that necessary ( maybe link to a docorg github issue if there is anything like that )

Reply:
Fixed.